src/Features/LanguageServer/ProtocolUnitTests/SemanticTokens/SemanticTokensEditsTests.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Xunit;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests.SemanticTokens
+{
+    public class SemanticTokensEditsTests : AbstractSemanticTokensTests
+    {
+        /*
+         * Markup for basic test case:
+         *     // Comment
+         *     static class C { }  
+         */
+        private static readonly int[] s_basicDataCase =
+            {
+                    // Line | Char | Len | Token type                                                             | Modifier
+                    0,        0,     10,   SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Comment), 0, // '// Comment'
+                    1,        0,     6,    SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Keyword), 0, // 'static'
+                    0,        7,     5,    SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Keyword), 0, // 'class'
+                    0,        6,     1,    SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Class), (int)TokenModifiers.Static, // 'C'
+                    0,        2,     1,    SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Operator), 0, // '{'
+                    0,        2,     1,    SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Operator), 0, // '}'
+            };
+
+        /*
+         * Markup for single line test case:
+         *     // Comment
+         */
+        private static readonly int[] s_singleLineDataCase =
+            {
+                    // Line | Char | Len | Token type                                                             | Modifier
+                    0,        0,     10,   SemanticTokensHelpers.GetTokenTypeIndex(LSP.SemanticTokenTypes.Comment), 0, // '// Comment'
+            };
+
+        /// <summary>
+        /// Tests inserting a newline in the middle of the file.
+        /// </summary>
+        [Fact]
+        public async Task TestGetSemanticTokensEditsBasicAsync()
```

Comment:
instead of a doc comment, remove that, and call this TestInsertingNewLineInMiddleOfFile.